contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -175,22 +186,60 @@ abstract contract PoSValidatorManager is
             "PoSValidatorManager: minimum stake duration not met"
         );
 
+        // Uptime proofs include the absolute number of seconds the validator has been active.
+        uint64 uptimeSeconds;
+        if (includeUptimeProof) {
+            uptimeSeconds = _updateUptime(validationID, messageIndex);
+        } else {
+            uptimeSeconds = $._validationUptimeSeconds[validationID];
+        }
+
+        uint256 reward = $._rewardCalculator.calculateReward({
+            stakeAmount: weightToValue(validator.startingWeight),
+            validatorStartTime: validator.startedAt,
+            stakingStartTime: validator.startedAt,
+            stakingEndTime: validator.endedAt,
+            uptimeSeconds: uptimeSeconds,
+            initialSupply: 0,
+            endSupply: 0
+        });
+        require(reward > 0, "PoSValidatorManager: validation ineligble for rewards");
+        $._redeemableValidatorRewards[validationID] += reward;
+    }
+
+    function forceInitializeEndValidation(
```

Comment:
We could combine this and `initializeEndValidation` by making an internal function that returns `reward`, and then checking it in one but not the other.